subsys/trusted_storage/src/auth_crypt_backend.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/util.h>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+
+#include "settings_helpers.h"
+#include "auth_crypt_backend.h"
+#include "auth_crypt_nonce.h"
+
+/* AEAD NONCE suffix */
+#define TRUSTED_STORAGE_FILENAME_SUFFIX_NONCE ".nonce"
+
+/*
+ * AEAD based Authenticated Encrypted trust implementation
+ *
+ * Actual implementation uses:
+ * - Hexadecimal UID imported as KEY
+ * - UID+Flags+Size as additional parameter
+ * - NONCE is an incrementing number at each encryption
+ * - TAG is left at the end of output data
```

Comment:
tag is the before output data other places?

Reply:
In the mbedTLS implementation, it looks like the tag comes before the encrypted data,
But do you think we can remove that here?